integration/internal/integration_tests/apiserver_integration_test.go

```diff
@@ -0,0 +1,24 @@
+package integration_tests
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/kubernetes-sig-testing/frameworks/integration"
+)
+
+var _ = Describe("APIServer", func() {
```

Comment:
That's an awful lot of code to guard against a nil value.  I'm not sure ginkgo is a good fit for unit testing - too heavyweight.

Reply:
OK -- I see what you mean!

I'd like to add a bit of background on why this test exists in the current form:
- The thing panic'ed on me, so I wrote the test which tests against exactly that
- While implementing the fix, I introduced that error and I wanted the test to actually check for that error -- an not allow any error which could potentially mask a different problem
- We are just used to using ginkgo & gomega

I tried to implement the [same test in 3 different ways](https://gist.github.com/hoegaarden/b2331a481a46ef017d3d7e99d190d95c) and those are my **personal**, **super-subjective** takeaways:
- Regarding SLOC it does not seem to make that much of a difference. In this example at least.
- I think, the example with ginkgo & gomega reads the nicest.

Having said that I'd be happy to see an example of how you'd picture this test -- I am happy to learn and adapt!

Reply:
I think it would be preferable to move validation like this out of the start method.  This would simplify testing sine there would be no need to check for panic.

```golang
func (s *APIServer) Start() error {
  if err := s.validateStart() {
    return error
  }
}
```